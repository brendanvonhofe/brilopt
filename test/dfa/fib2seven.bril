# ARGS: reach
@fib(x: int): int {
.zerocase:
  out: int = const 0;
  zero: int = const 0;
  iszero: bool = eq x zero;
  br iszero .ret .onecase;
.onecase:
  out: int = const 1;
  one: int = const 1;
  isone: bool = eq x one;
  br isone .ret .body;
.body:
  two: int = const 2;
  xmone: int = sub x one;
  xmtwo: int = sub x two;
  fibone: int = call @fib xmone;
  fibtwo: int = call @fib xmtwo;
  out: int = add fibone fibtwo;
.ret:
  ret out;
}
@main {
  five: int = const 7;
  fibfive: int = call @fib five;
  print fibfive;
}
