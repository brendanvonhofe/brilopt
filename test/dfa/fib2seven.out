entry:
  in:  
  out: 
zerocase:
  in:  
  out: iszero_zerocase_3 out_zerocase_1 zero_zerocase_2
onecase:
  in:  iszero_zerocase_3 out_zerocase_1 zero_zerocase_2
  out: isone_onecase_3 iszero_zerocase_3 one_onecase_2 out_onecase_1 zero_zerocase_2
body:
  in:  isone_onecase_3 iszero_zerocase_3 one_onecase_2 out_onecase_1 zero_zerocase_2
  out: fibone_body_4 fibtwo_body_5 isone_onecase_3 iszero_zerocase_3 one_onecase_2 out_body_6 two_body_1 xmone_body_2 xmtwo_body_3 zero_zerocase_2
ret:
  in:  fibone_body_4 fibtwo_body_5 isone_onecase_3 iszero_zerocase_3 one_onecase_2 out_body_6 out_onecase_1 out_zerocase_1 two_body_1 xmone_body_2 xmtwo_body_3 zero_zerocase_2
  out: fibone_body_4 fibtwo_body_5 isone_onecase_3 iszero_zerocase_3 one_onecase_2 out_body_6 out_onecase_1 out_zerocase_1 two_body_1 xmone_body_2 xmtwo_body_3 zero_zerocase_2
exit:
  in:  fibone_body_4 fibtwo_body_5 isone_onecase_3 iszero_zerocase_3 one_onecase_2 out_body_6 out_onecase_1 out_zerocase_1 two_body_1 xmone_body_2 xmtwo_body_3 zero_zerocase_2
  out: fibone_body_4 fibtwo_body_5 isone_onecase_3 iszero_zerocase_3 one_onecase_2 out_body_6 out_onecase_1 out_zerocase_1 two_body_1 xmone_body_2 xmtwo_body_3 zero_zerocase_2

entry:
  in:  
  out: 
main1:
  in:  
  out: fibfive_main1_1 five_main1_0
exit:
  in:  fibfive_main1_1 five_main1_0
  out: fibfive_main1_1 five_main1_0

